src/fu-util.c

```diff
@@ -2396,6 +2396,165 @@ fu_util_security (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_upload_security_build_metadata (JsonBuilder *builder, GError **error)
+{
+	g_autoptr(GHashTable) hash = NULL;
+	struct {
+		const gchar *key;
+		const gchar *val;
+	} distro_kv[] = {
+		{ "ID",			"DistroId" },
+		{ "VERSION_ID",		"DistroVersion" },
+		{ "VARIANT_ID",		"DistroVariant" },
+		{ NULL, NULL }
+	};
+
+	/* get all required os-release keys */
+	hash = fwupd_get_os_release (error);
+	if (hash == NULL)
+		return FALSE;
+	for (guint i = 0; distro_kv[i].key != NULL; i++) {
+		const gchar *tmp = g_hash_table_lookup (hash, distro_kv[i].key);
+		if (tmp != NULL) {
+			json_builder_set_member_name (builder, distro_kv[i].val);
+			json_builder_add_string_value (builder, tmp);
+		}
+	}
```

Comment:
So I think for analyzing how accurate things are you'll probably want `/proc/cmdline` too.

Reply:
Do we also want that for "normal" firmware reports?

Reply:
I don't think it's needed there.  The reason for this is if someone has HSI:0 and you see they have `iommu=off`, potentially LVFS can either filter this or return back a message about something they can do.

Reply:
Hmm, there's quite a lot of personal-ish data there. 

For instance:

`BOOT_IMAGE=(hd2,gpt2)/vmlinuz-5.6.11-300.fc32.x86_64 root=UUID=01234dff-c7d0-4e29-b312-888e710ec39b ro resume=/dev/mapper/luks-012345-5c97-427c-87d8-e6c46c662562 rd.luks.uuid=luks-012345-5c97-427c-87d8-e6c46c662562 rhgb quiet`

I'm wondering if we should split it by spaces in the client tool and then whitelist/backlist some stuff. For instance, blacklisting `BOOT_IMAGE`, `root`, `resume`, `rd.luks.uuid` etc

Reply:
Yes I think that makes sense to filter out some of those things we don't care about and could be sensitive.  I would prefer to blacklist items than whitelist, because we don't know yet what we don't know in terms of what people are doing that decreases the security on their system from `/proc/cmdline`.

Reply:
Okay, added.